mlir/lib/Dialect/Tosa/Transforms/TosaValidation.cpp

```diff
@@ -543,6 +543,10 @@ bool TosaValidation::isValidElementType(Type type) {
 void TosaValidation::runOnOperation() {
   configLevelAndProfile();
   getOperation().walk([&](Operation *op) {
+    if (!op->getDialect() ||
+        op->getDialect()->getNamespace() != TosaDialect::getDialectNamespace())
```

Comment:
This isn't an efficient way to achieve this kind of comparison (and it's gonna be done on every operation in the IR). The best way to do is likely:

```
void TosaValidation::runOnOperation() {
   configLevelAndProfile();
   TosaDialect tosaDialect = getContext()->getLoadedDialect<TosaDialect>();
   if (!tosaDialect) return;
   
   getOperation().walk([&](Operation *op) {
     if (op->getDialect()  != tosaDialect)
```

Reply:
Thanks for the suggestion, I've uploaded #120960 to address this